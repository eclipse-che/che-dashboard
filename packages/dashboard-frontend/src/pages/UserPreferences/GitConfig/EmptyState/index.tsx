/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { EmptyState, EmptyStateIcon, EmptyStateVariant, Title } from '@patternfly/react-core';
import { CogIcon } from '@patternfly/react-icons';
import React from 'react';

export class GitConfigEmptyState extends React.PureComponent {
  public render(): React.ReactElement {
    return (
      <EmptyState isFullHeight={true} variant={EmptyStateVariant.small}>
        <EmptyStateIcon icon={CogIcon} />
        <Title headingLevel="h4" size="lg">
          No gitconfig found
        </Title>
      </EmptyState>
    );
  }
}
