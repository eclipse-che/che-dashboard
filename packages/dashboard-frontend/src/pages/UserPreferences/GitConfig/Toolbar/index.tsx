/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import {
  Button,
  ButtonVariant,
  Toolbar,
  ToolbarContent,
  ToolbarItem,
} from '@patternfly/react-core';
import { CodeIcon, ListIcon, PlusCircleIcon } from '@patternfly/react-icons';
import React from 'react';

export type Props = {
  mode: 'form' | 'viewer';
  onAdd: () => void;
  onChangeMode: (mode: 'form' | 'viewer') => void;
};

export class GitConfigToolbar extends React.PureComponent<Props> {
  private handleModeChange(mode: 'form' | 'viewer'): void {
    this.setState({ mode });
    this.props.onChangeMode(mode);
  }

  public render(): React.ReactElement {
    const { mode } = this.props;

    return (
      <Toolbar style={{ paddingBottom: 0 }}>
        <ToolbarContent>
          <ToolbarItem alignment={{ default: 'alignRight' }}>
            {mode === 'form' ? (
              <Button
                variant={ButtonVariant.link}
                icon={<CodeIcon />}
                iconPosition="left"
                onClick={() => this.handleModeChange('viewer')}
              >
                Switch to Viewer
              </Button>
            ) : (
              <Button
                variant={ButtonVariant.link}
                icon={<ListIcon />}
                iconPosition="left"
                onClick={() => this.handleModeChange('form')}
              >
                Switch to Form
              </Button>
            )}
            <Button
              variant={ButtonVariant.link}
              icon={<PlusCircleIcon />}
              iconPosition="left"
              onClick={() => this.props.onAdd()}
            >
              Import Git Configuration
            </Button>
          </ToolbarItem>
        </ToolbarContent>
      </Toolbar>
    );
  }
}
