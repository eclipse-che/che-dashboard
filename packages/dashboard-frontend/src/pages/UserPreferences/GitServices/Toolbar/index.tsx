/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import {
  Button,
  ButtonVariant,
  Toolbar,
  ToolbarContent,
  ToolbarItem,
} from '@patternfly/react-core';
import React from 'react';

import { IGitOauth } from '@/store/GitOauthConfig/types';

export type Props = {
  isDisabled: boolean;
  selectedItems: IGitOauth[];
  onRevokeButton: () => void;
};

export class GitServicesToolbar extends React.PureComponent<Props> {
  public handleRevokeButtonClick(): void {
    this.props.onRevokeButton();
  }

  render(): React.ReactNode {
    const { isDisabled, selectedItems } = this.props;

    const buttonDisabled = isDisabled || selectedItems.length === 0;

    return (
      <React.Fragment>
        <Toolbar>
          <ToolbarContent>
            <ToolbarItem>
              <Button
                variant={ButtonVariant.danger}
                isDisabled={buttonDisabled}
                data-testid="bulk-revoke-button"
                onClick={() => this.handleRevokeButtonClick()}
              >
                Revoke
              </Button>
            </ToolbarItem>
          </ToolbarContent>
        </Toolbar>
      </React.Fragment>
    );
  }
}
