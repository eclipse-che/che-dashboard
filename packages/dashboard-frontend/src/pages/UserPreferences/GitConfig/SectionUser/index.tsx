/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Form, FormSection, Panel, PanelMain, PanelMainBody } from '@patternfly/react-core';
import * as React from 'react';

import { GitConfigUserEmail } from '@/pages/UserPreferences/GitConfig/SectionUser/Email';
import { GitConfigUserName } from '@/pages/UserPreferences/GitConfig/SectionUser/Name';
import { GitConfigUser } from '@/store/GitConfig';

export type Props = {
  config: GitConfigUser;
  onChange: (gitConfigUser: GitConfigUser) => void;
};

export class GitConfigSectionUser extends React.PureComponent<Props> {
  private handleChange(partialConfig: Partial<GitConfigUser>): void {
    const { config, onChange } = this.props;

    onChange({
      ...config,
      ...partialConfig,
    });
  }

  public render(): React.ReactElement {
    const { config } = this.props;
    return (
      <Panel>
        <PanelMain>
          <PanelMainBody>
            <Form isHorizontal onSubmit={e => e.preventDefault()}>
              <FormSection title="[user]" label="user">
                <GitConfigUserName
                  value={config.name}
                  onChange={name => this.handleChange({ name })}
                />
                <GitConfigUserEmail
                  value={config.email}
                  onChange={email => this.handleChange({ email })}
                />
              </FormSection>
            </Form>
          </PanelMainBody>
        </PanelMain>
      </Panel>
    );
  }
}
