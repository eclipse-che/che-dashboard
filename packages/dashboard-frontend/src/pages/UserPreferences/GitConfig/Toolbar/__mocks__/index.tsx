/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitConfig/Toolbar';

export class GitConfigToolbar extends React.PureComponent<Props> {
  render() {
    const { mode, onAdd, onChangeMode } = this.props;
    const nextMode = mode === 'form' ? 'viewer' : 'form';
    return (
      <div>
        Mock Git Config Toolbar
        <span data-testid="toolbar-mode">{mode}</span>
        <button data-testid="toolbar-on-add" onClick={onAdd}>
          Add
        </button>
        <button data-testid="toolbar-on-change-mode" onClick={() => onChangeMode(nextMode)}>
          Switch Mode
        </button>
      </div>
    );
  }
}
