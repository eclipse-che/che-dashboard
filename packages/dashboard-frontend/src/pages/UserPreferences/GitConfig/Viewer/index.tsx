/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { api } from '@eclipse-che/common';
import { PageSection } from '@patternfly/react-core';
import * as ini from 'multi-ini';
import React from 'react';

import { BasicViewer } from '@/components/BasicViewer';
import { GitConfig } from '@/store/GitConfig';

export type Props = {
  config: GitConfig;
};

export class GitConfigViewer extends React.PureComponent<Props> {
  /**
   * @throws
   * Serializes `gitConfig` object.
   */
  private fromGitConfig(gitConfig: api.IGitConfig['gitconfig']): string {
    const serializer = new ini.Serializer();
    const gitconfigStr = serializer
      .serialize(gitConfig)
      .replace(/\n/g, '\n    ')
      .replace(/ {4}\[/g, '[');
    return gitconfigStr;
  }

  render() {
    const { config } = this.props;

    const gitConfigStr = this.fromGitConfig(config);

    return (
      <PageSection>
        <BasicViewer value={gitConfigStr} id="gitconfig-editor-id" />
      </PageSection>
    );
  }
}
