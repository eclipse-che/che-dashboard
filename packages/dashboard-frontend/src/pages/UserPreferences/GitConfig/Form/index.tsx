/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Button, Flex, Form, PageSection, PageSectionVariants } from '@patternfly/react-core';
import React, { FormEvent } from 'react';

import { GitConfigSectionUser } from '@/pages/UserPreferences/GitConfig/Form/SectionUser';
import * as GitConfigStore from '@/store/GitConfig';

export type Props = {
  isLoading: boolean;
  gitConfig: GitConfigStore.GitConfig;
  onSave: (gitConfig: GitConfigStore.GitConfig) => Promise<void>;
  onReload: () => Promise<void>;
};
export type State = {
  isValid: boolean;
  nextGitConfig: GitConfigStore.GitConfig | undefined;
};

export class GitConfigForm extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      isValid: true,
      nextGitConfig: undefined,
    };
  }

  private handleSubmit(e: FormEvent): void {
    e.preventDefault();
    e.stopPropagation();
  }

  private async handleSave(): Promise<void> {
    const nextGitConfig = {
      ...this.props.gitConfig,
      ...(this.state.nextGitConfig || {}),
    };
    await this.props.onSave(nextGitConfig);
  }

  private async handleReload(): Promise<void> {
    try {
      await this.props.onReload();
      this.setState({
        nextGitConfig: undefined,
        isValid: true,
      });
    } catch (e) {
      // ignore
    }
  }

  private handleChangeConfig(gitConfig: GitConfigStore.GitConfig, isValid: boolean): void {
    this.setState({
      nextGitConfig: gitConfig,
      isValid,
    });
  }

  public render(): React.ReactElement {
    const { gitConfig, isLoading } = this.props;
    const { isValid, nextGitConfig } = this.state;

    const config = { ...gitConfig, ...(nextGitConfig || {}) };
    const isSaveDisabled = isLoading || isValid === false || nextGitConfig === undefined;

    return (
      <PageSection variant={PageSectionVariants.light}>
        <Form isHorizontal onSubmit={e => this.handleSubmit(e)}>
          <GitConfigSectionUser
            isLoading={isLoading}
            config={config}
            onChange={(gitConfig, isValid) => this.handleChangeConfig(gitConfig, isValid)}
          />
          <Flex>
            <Button
              data-testid="button-save"
              isDisabled={isSaveDisabled}
              type="button"
              variant="primary"
              onClick={() => this.handleSave()}
            >
              Save
            </Button>
            <Button
              data-testid="button-reload"
              disabled={isLoading}
              isDisabled={isLoading}
              type="reset"
              variant="secondary"
              onClick={() => this.handleReload()}
            >
              Reload
            </Button>
          </Flex>
        </Form>
      </PageSection>
    );
  }
}
