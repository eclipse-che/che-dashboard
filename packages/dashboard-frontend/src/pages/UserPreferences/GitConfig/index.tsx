/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { helpers } from '@eclipse-che/common';
import { AlertVariant, PageSection } from '@patternfly/react-core';
import React from 'react';
import { connect, ConnectedProps } from 'react-redux';

import ProgressIndicator from '@/components/Progress';
import { lazyInject } from '@/inversify.config';
import { GitConfigEmptyState } from '@/pages/UserPreferences/GitConfig/EmptyState';
import { GitConfigForm } from '@/pages/UserPreferences/GitConfig/Form';
import { AppAlerts } from '@/services/alerts/appAlerts';
import { AppState } from '@/store';
import * as GitConfigStore from '@/store/GitConfig';
import {
  selectGitConfig,
  selectGitConfigError,
  selectGitConfigIsLoading,
} from '@/store/GitConfig/selectors';

export type Props = MappedProps;

class GitConfig extends React.PureComponent<Props> {
  @lazyInject(AppAlerts)
  private readonly appAlerts: AppAlerts;

  public async componentDidMount(): Promise<void> {
    const { gitConfigIsLoading } = this.props;

    if (gitConfigIsLoading === true) {
      this.init();
      return;
    }

    await this.handleReload();
  }

  public componentDidUpdate(prevProps: Props): void {
    this.init(prevProps);
  }

  private init(prevProps?: Props): void {
    const { gitConfigError } = this.props;
    const prevGitConfigError = prevProps?.gitConfigError;
    if (gitConfigError && gitConfigError !== prevGitConfigError) {
      this.appAlerts.showAlert({
        key: 'gitconfig-error',
        title: helpers.errors.getMessage(gitConfigError),
        variant: AlertVariant.danger,
      });
    } else if (gitConfigError === undefined && gitConfigError !== prevGitConfigError) {
      this.appAlerts.removeAlert('gitconfig-error');
    }
  }

  private async handleSave(gitConfig: GitConfigStore.GitConfig): Promise<void> {
    try {
      await this.props.updateGitConfig(gitConfig);
      this.appAlerts.showAlert({
        key: 'gitconfig-success',
        title: 'Gitconfig saved successfully.',
        variant: AlertVariant.success,
      });
    } catch (error) {
      console.error('Failed to update gitconfig', error);
    }
  }

  private async handleReload(): Promise<void> {
    try {
      await this.props.requestGitConfig();
    } catch (error) {
      console.error('Failed to reload gitconfig', error);
    }
  }

  public render(): React.ReactElement {
    const { gitConfigIsLoading, gitConfig } = this.props;

    const isEmpty = gitConfig === undefined;

    return (
      <React.Fragment>
        <ProgressIndicator isLoading={gitConfigIsLoading} />
        {isEmpty ? (
          <GitConfigEmptyState />
        ) : (
          <PageSection>
            <GitConfigForm
              gitConfig={gitConfig}
              isLoading={gitConfigIsLoading}
              onSave={async gitConfig => await this.handleSave(gitConfig)}
              onReload={async () => await this.handleReload()}
            />
          </PageSection>
        )}
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state: AppState) => ({
  gitConfig: selectGitConfig(state),
  gitConfigIsLoading: selectGitConfigIsLoading(state),
  gitConfigError: selectGitConfigError(state),
});

const connector = connect(mapStateToProps, GitConfigStore.actionCreators, null, {
  // forwardRef is mandatory for using `@react-mock/state` in unit tests
  forwardRef: true,
});

type MappedProps = ConnectedProps<typeof connector>;
export default connector(GitConfig);
