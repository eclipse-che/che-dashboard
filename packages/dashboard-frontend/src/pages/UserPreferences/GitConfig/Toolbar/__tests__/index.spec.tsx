/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import userEvent from '@testing-library/user-event';
import React from 'react';

import { GitConfigToolbar } from '@/pages/UserPreferences/GitConfig/Toolbar';
import getComponentRenderer, { screen, waitFor } from '@/services/__mocks__/getComponentRenderer';

const { createSnapshot, renderComponent } = getComponentRenderer(getComponent);

const mockOnAdd = jest.fn();
const mockOnChangeMode = jest.fn();

describe('GitConfigToolbar', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('snapshot, form mode', () => {
    const snapshot = createSnapshot('form');
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('snapshot, viewer mode', () => {
    const snapshot = createSnapshot('form');
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('import Git Configuration', () => {
    renderComponent('form');

    expect(mockOnAdd).not.toHaveBeenCalled();

    const button = screen.getByRole('button', { name: 'Import Git Configuration' });
    button.click();

    expect(mockOnAdd).toHaveBeenCalled();
  });

  test('change mode to viewer', async () => {
    renderComponent('form');

    expect(mockOnChangeMode).not.toHaveBeenCalled();
    expect(screen.queryByRole('button', { name: 'Switch to Form' })).toBeNull();

    const viewerButton = screen.getByRole('button', { name: 'Switch to Viewer' });
    await userEvent.click(viewerButton);

    await waitFor(() => expect(mockOnChangeMode).toHaveBeenCalledWith('viewer'));
  });

  test('change mode to form', async () => {
    renderComponent('viewer');

    expect(mockOnChangeMode).not.toHaveBeenCalled();
    expect(screen.queryByRole('button', { name: 'Switch to Viewer' })).toBeNull();

    const formButton = screen.getByRole('button', { name: 'Switch to Form' });
    await userEvent.click(formButton);

    await waitFor(() => expect(mockOnChangeMode).toHaveBeenCalledWith('form'));
  });
});

function getComponent(mode: 'form' | 'viewer'): React.ReactElement {
  return <GitConfigToolbar mode={mode} onAdd={mockOnAdd} onChangeMode={mockOnChangeMode} />;
}
