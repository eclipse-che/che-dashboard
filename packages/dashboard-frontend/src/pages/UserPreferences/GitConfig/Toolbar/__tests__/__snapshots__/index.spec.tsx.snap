// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitConfigToolbar snapshot, form mode 1`] = `
<div
  className="pf-c-toolbar"
  data-ouia-component-id="OUIA-Generated-Toolbar-1"
  data-ouia-component-type="PF4/Toolbar"
  data-ouia-safe={true}
  id="pf-random-id-0"
  style={
    {
      "paddingBottom": 0,
    }
  }
>
  <div
    className="pf-c-toolbar__content"
  >
    <div
      className="pf-c-toolbar__content-section"
    >
      <div
        className="pf-c-toolbar__item pf-m-align-right"
      >
        <button
          aria-disabled={false}
          aria-label={null}
          className="pf-c-button pf-m-link"
          data-ouia-component-id="OUIA-Generated-Button-link-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe={true}
          disabled={false}
          onClick={[Function]}
          role={null}
          type="button"
        >
          <span
            className="pf-c-button__icon pf-m-start"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 640 512"
              width="1em"
            >
              <path
                d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
              />
            </svg>
          </span>
          Switch to Viewer
        </button>
        <button
          aria-disabled={false}
          aria-label={null}
          className="pf-c-button pf-m-link"
          data-ouia-component-id="OUIA-Generated-Button-link-2"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe={true}
          disabled={false}
          onClick={[Function]}
          role={null}
          type="button"
        >
          <span
            className="pf-c-button__icon pf-m-start"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
              />
            </svg>
          </span>
          Import Git Configuration
        </button>
      </div>
    </div>
    <div
      className="pf-c-toolbar__expandable-content"
      id="pf-random-id-0-expandable-content-0"
    >
      <div
        className="pf-c-toolbar__group"
      />
    </div>
  </div>
  <div
    className="pf-c-toolbar__content pf-m-hidden"
    hidden={true}
  >
    <div
      className="pf-c-toolbar__group"
    />
  </div>
</div>
`;

exports[`GitConfigToolbar snapshot, viewer mode 1`] = `
<div
  className="pf-c-toolbar"
  data-ouia-component-id="OUIA-Generated-Toolbar-2"
  data-ouia-component-type="PF4/Toolbar"
  data-ouia-safe={true}
  id="pf-random-id-1"
  style={
    {
      "paddingBottom": 0,
    }
  }
>
  <div
    className="pf-c-toolbar__content"
  >
    <div
      className="pf-c-toolbar__content-section"
    >
      <div
        className="pf-c-toolbar__item pf-m-align-right"
      >
        <button
          aria-disabled={false}
          aria-label={null}
          className="pf-c-button pf-m-link"
          data-ouia-component-id="OUIA-Generated-Button-link-3"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe={true}
          disabled={false}
          onClick={[Function]}
          role={null}
          type="button"
        >
          <span
            className="pf-c-button__icon pf-m-start"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 640 512"
              width="1em"
            >
              <path
                d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
              />
            </svg>
          </span>
          Switch to Viewer
        </button>
        <button
          aria-disabled={false}
          aria-label={null}
          className="pf-c-button pf-m-link"
          data-ouia-component-id="OUIA-Generated-Button-link-4"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe={true}
          disabled={false}
          onClick={[Function]}
          role={null}
          type="button"
        >
          <span
            className="pf-c-button__icon pf-m-start"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
              />
            </svg>
          </span>
          Import Git Configuration
        </button>
      </div>
    </div>
    <div
      className="pf-c-toolbar__expandable-content"
      id="pf-random-id-1-expandable-content-1"
    >
      <div
        className="pf-c-toolbar__group"
      />
    </div>
  </div>
  <div
    className="pf-c-toolbar__content pf-m-hidden"
    hidden={true}
  >
    <div
      className="pf-c-toolbar__group"
    />
  </div>
</div>
`;
